### PR TITLE
upgrade apicast and system sidekiq prometheus rules

### PR DESCRIPTION
**Pull request by eguzki**
I have compared monitoring resources differences between 2.9 and 2.10 that require upgrading procedure not implemented currently. I found:
- [x] Apicast prometheus rules updated
- [x] System Sidekiq prometheus rules updated

### CODE REVIEW

File: pkg/3scale/amp/operator/upgrade.go [hunk 1]
```diff
@@ -424,6 +425,20 @@ func (u *UpgradeApiManager) upgradeMonitoringSettings() (reconcile.Result, error
 	}
 	updated = updated || updatedTmp
 
+	// Apicast rules
+	updatedTmp, err = u.upgradePrometheusRules(component.ApicastPrometheusRules(u.apiManager.Namespace))
```
Comment (miguelsorianod) [inline pkg/3scale/amp/operator/upgrade.go:429]: With the current implementation it seems there's no real good solution for a problem like this:

If we upgrade or reconcile Prometheus Rules, possible changes that users might have done will be undone. As @slopezz commented in https://github.com/3scale/3scale-operator/pull/495#issuecomment-713660615 there's a very real possibility users might be doing that for Prometheus Rules, although as far as I know we have never really told/documented that that can be changed so far.
If we don't upgrade nor reconcile Prometheus Rules then the users will not have newer changes that we do on the Prometheus rules.

Comment (eguzki) [reply pkg/3scale/amp/operator/upgrade.go:429]: We could implement upgrade process to add only new rules. Do not reconcile entire content, only new content. Of course, if there are conflicts like: customer updates rule A, new release also updates rule A, then there is nothing the operator can do.

Comment (miguelsorianod) [reply pkg/3scale/amp/operator/upgrade.go:429]: If we only add new content then updated/fixed content of existing alerts will be not incorporated.

I think at this moment with the current implementation the options we have are reverting possible manually made changes by the users at upgrade time or not providing those updated/fixed changes. It is basically a trade-off.

**Comment by eguzki** [comment]
We have decided to implement the upgrade of the prometheus rules, overwriting any custom update the customer could have done.

**Review by miguelsorianod — APPROVED** [review]

**Pull request merged**